c2corg_api/ext/colander_ext.py

```diff
@@ -53,11 +53,16 @@ def deserialize(self, node, cstruct):
         if cstruct is null or cstruct == '':
             return null
         try:
-            geojson = json.loads(cstruct)
-            geom_type = geojson['type'].upper()
+            geojson_data = json.loads(cstruct)
+            geom_type = geojson_data['type'].upper()
             allowed_types = self.geometry_types
             if geom_type in allowed_types:
-                return wkbelement_from_geojson(geojson, self.srid)
+                validation = geojson.is_valid(geojson_data)
```

Comment:
> It seems that geojson.is_valid only accepts 2D and 3D geometries, not 4D:

In that case (testing a Point geometry), is it really a problem?

According to my tests the problem is rather that 4D multilinestrings are detected as MultiPoint (?!) whose points are indeed rejected because not compatible with the 4D:
https://github.com/frewsxcv/python-geojson/blob/master/geojson/validation.py#L32-L33

I will soon upgrade my PR with "custom" validation.